org-formation/650-identity-providers/github-oidc-provider-access.njk

```diff
@@ -89,12 +89,15 @@ Resources:
             Condition:
               StringEquals:
                 token.actions.githubusercontent.com:aud: !Ref ClientIdList
-  {% if Repository.branch == '*' %}
               StringLike:
-                token.actions.githubusercontent.com:sub: !Sub repo:${GitHubOrg}/{{ Repository.name }}:{{Repository.branch }}
+                token.actions.githubusercontent.com:sub: [
+  {% if Repository.branch == '*' %}
+                  "!Sub repo:${GitHubOrg}/{{ Repository.name }}:{{Repository.branch }}",
   {% else %}
-                token.actions.githubusercontent.com:sub: !Sub repo:${GitHubOrg}/{{ Repository.name }}:ref:refs/heads/{{ Repository.branch }}
+                  "!Sub repo:${GitHubOrg}/{{ Repository.name }}:ref:refs/heads/{{ Repository.branch }}",
   {% endif %}
+                  "!Sub repo:${GitHubOrg}/{{ Repository.name }}:ref:refs/tags/*"
```

Comment:
Only wildcards '*' and '?' allowed from what I see, used '*' since this is shared with other repos.

Reply:
this is not gonna work.  `StringEquals` and `StringLike` are dict objects, it cannot take a list of conditions.    you can test it out yourself by editing the trust policy in the AWS console. Furthermore AWS does not allow adding multiple `StringEquals` and `StringLike` conditions in a policy.

Reply:
It's not a list of conditions, it's a list of values for 'token.actions.githubusercontent.com:sub'. As I understand they should be ORed. In the end I'm just adding a value to the existing StringLike clause, am I not?
Conditions:
  StringEquals:
    token.actions.githubusercontent.com:aud: !Ref ClientIdList
  StringLike: [
    "!Sub repo:${GitHubOrg}/{{ Repository.name }}:ref:refs/heads/{{ Repository.branch }}",
    "!Sub repo:${GitHubOrg}/{{ Repository.name }}:ref:refs/tags/*"
  ]

Reply:
ahh i see.  the list is inside of `token.actions.githubusercontent.com:sub:` which does work.  so it's something like this?
```
"Condition": {
	"StringEquals": {
		"token.actions.githubusercontent.com:aud": "sts.amazonaws.com"
	},
	"StringLike": {
		"token.actions.githubusercontent.com:sub": ["repo:Sage-Bionetworks/aws-infra:ref:refs/heads/master",
		"repo:Sage-Bionetworks/aws-infra:ref:refs/tags/*"]
	}
}
```

Reply:
Yes, sorry I see now I forgot the key in the StringLike...